hpcgap/src/c_type1.c

```diff
@@ -1083,17 +1087,23 @@ static Obj  HdlrFunc11 (
   C_ELM_LIST_FPL( t_1, l_cache, l_hash )
   l_cached = t_1;
   
-  /* if IS_EQUAL_FLAGS( flags, cached![2] ) then */
+  /* if IS_EQUAL_FLAGS( flags, cached![POS_FLAGS_TYPE] ) then */
   t_3 = GF_IS__EQUAL__FLAGS;
-  C_ELM_POSOBJ_NLE( t_4, l_cached, 2 );
+  t_5 = GC_POS__FLAGS__TYPE;
+  CHECK_BOUND( t_5, "POS_FLAGS_TYPE" )
+  CHECK_INT_SMALL_POS( t_5 )
+  C_ELM_POSOBJ_NLE( t_4, l_cached, INT_INTOBJ(t_5) );
```

Comment:
Hmmpf, looking at this, the compiler does not pick up these constants and generates suboptimal code as a result. The reason is that gac does not load the GAP library. To fix this, I'll have to move the constants to the kernel after all *sigh*. Will do that later.